### ISSUE TITLE
IPv6 routes are not imported into conf.rout6 if the destination is not in same subnet as interface address

### ISSUE THREAD

**Issue by bharathgrid** [secdev/scapy#994]
Scapy Version: 2.3.3
System: Fedora 15
Python Version: 2.7.1

Issue: Some of the IPv6 routes added in linux are not imported into conf.route6 file if the destination is not in same subnet as any of the interface IPv6 addresses

Ex:
[root@admin]# ip -6 route
64:ff9b::/32 via 2001:abcd::1 dev eth0  metric 1024
1001:abcd::/32 via 2001:abcd::2 dev eth0  metric 1024
2001:abcd::707:707 dev eth0  proto kernel  metric 256
4001:abcd::1 dev eth0  metric 1024
4001:abcd::/32 via 2001:abcd::1 dev eth0  metric 1024

[root@admin]# ip -6 addr show eth0
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9000 qlen 1000
    inet6 2001:abcd::707:707/128 scope global
       valid_lft forever preferred_lft forever
    inet6 2001:abcd::7:707:700:0/64 scope global
       valid_lft forever preferred_lft forever
    inet6 2001:abcd::707:707:0/32 scope global
       valid_lft forever preferred_lft forever
    inet6 2001:abcd:7ee::707:707:ee/32 scope global
       valid_lft forever preferred_lft forever

       
![capture](https://user-images.githubusercontent.com/6072360/34089458-03e2a24a-e365-11e7-9de0-5bc5c354aeb7.JPG)

From code: In 'arch/linux.py'  file, 'read_routes6' API

cset = scapy.utils6.construct_source_candidate_set(d, dp, devaddrs)  --> fetches a non-zero set only if destination and devaddrs have same prefix.

Sorry if I this is not an issue/already known issue.

**Comment by guedou** [comment]
Please try to reproduce this issue with the current github version.

**Comment by bharathgrid** [comment]
Hi guedou,

Thanks for looking into the issue. On further debugging I found that the routes are not added because the destination IP addresses used in this case doesn't fall under Global unicast (2000::/3) or local (fc00::/7)  address space. 

Since my testbed topology is local I was using addresses without considering the RFC specifications.

Closing the thread as it is not a valid issue.

Thanks again for looking into it.
